src/dataflow-types/src/client.rs

```diff
@@ -131,7 +134,14 @@ impl ComputeCommandKind {
 )]
 pub enum StorageCommand {
     /// Create the enumerated sources, each associated with its identifier.
-    CreateSources(Vec<(GlobalId, crate::types::sources::SourceDesc)>),
+    ///
+    /// For each identifier, there is a source description and a valid `since` frontier.
+    CreateSources(
+        Vec<(
+            GlobalId,
+            (crate::types::sources::SourceDesc, Antichain<Timestamp>),
```

Comment:
This seems like another moment where a named type would be nice but no need to address right now!

Reply:
Yeah, point taken. It's a thing that should vanish eventually (STORAGE should own determining the initial `since` frontier) so I can't bring myself to make it look nice and clean.

Reply:
👍🏽 works for me